### PR TITLE
fix: Table topPaginationNode display logic should same with bottomPaginationNode

### PR DESCRIPTION
**Pull request by zhangchen915**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

- topPaginationNode 和 bottomPaginationNode 的展示判断逻辑应该一致
- 改成了 mergedPagination?.total  判断是否展示分页

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix Table topPaginationNode display logic should same with bottomPaginationNode     |
| 🇨🇳 Chinese |  修复 Table 组件  topPaginationNode 与 bottomPaginationNode 的展示逻辑应该一致      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
来个测试用例？

**Comment by zombieJ** [comment]
Changelog 也需要写一下哈~

File: components/table/Table.tsx [hunk 1]
```diff
@@ -420,7 +420,7 @@ function Table<RecordType extends object = any>(props: TableProps<RecordType>) {
 
   let topPaginationNode: React.ReactNode;
   let bottomPaginationNode: React.ReactNode;
-  if (pagination !== false) {
+  if (pagination !== false && mergedPagination?.total) {
```
Comment (zombieJ) [inline components/table/Table.tsx:423]: 这边不用 `mergedData.length` 原因是？

Comment (zhangchen915) [reply components/table/Table.tsx:423]: https://github.com/ant-design/ant-design/issues/25836#issuecomment-671897664
mergedData 对于异步的数据应该还有问题

Comment (zombieJ) [reply components/table/Table.tsx:423]: 👌

**Pull request merged**